File: Java/Easy/0014-longest-common-prefix.java
class LongestCommonPrefix {
    public String longestCommonPrefix(String[] strs) {
        String res = strs[0];
        boolean isSameChar = false;
        for (int i = 0; i < strs.length; i++) {
            String temp = "";
            int j = 0;
            while (j < strs[i].length() && j < res.length()
                    && res.charAt(j) == strs[i].charAt(j)) {
                temp += strs[i].charAt(j);
                j++;
            }
            res = temp;
        }
        return res;
    }
}

File: JavaScript/Easy/0014-longest-common-prefix.js
/**
 * @param {string[]} strs
 * @return {string}
 */
var longestCommonPrefix = function(strs) {
    let res = strs[0];
    for (let i = 0; i < strs.length; i++) {
        let temp = "";
        let j = 0;
        while (j < strs[i].length && j < res.length && res.charAt(j) == strs[i].charAt(j) ) {
            temp += strs[i].charAt(j);
            j++;
        }
        res = temp;
    }
    return res;
};